examples/viewer/viewer.cc

```diff
@@ -307,7 +390,13 @@ static bool LoadObjAndConvert(float bmin[3], float bmax[3],
     for (size_t s = 0; s < shapes.size(); s++) {
       DrawObject o;
       std::vector<float> buffer;  // pos(3float), normal(3float), color(3float)
-      for (size_t f = 0; f < shapes[s].mesh.indices.size() / 3; f++) {
+
+	  // Check for smoothing group and compute smoothing normals
+	  std::map<int, float[3]> smoothVertexNormals;
+	  if (shapes[s].smoothingGroupId > 0)
+		  computeSmoothingNormals(attrib, shapes[s], smoothVertexNormals);
```

Comment:
If smoothing group id is > 0 for the shape, compute the smoothing normals for the vertexes of the shape.
The computed vertex normals are stored in smoothVertexNormals, which is a maps each vertex If of the shape to a normal.